test/e2e/storage/testsuites/volume_modify.go

```diff
@@ -0,0 +1,291 @@
+	type local struct {
+		config *storageframework.PerTestConfig
+
+		resource *storageframework.VolumeResource
+		vac      *storagev1alpha1.VolumeAttributesClass
+	}
+	var l local
+
+	// Beware that it also registers an AfterEach which renders f unusable. Any code using
+	// f must run inside an It or Context callback.
+	f := framework.NewFrameworkWithCustomTimeouts("volume-modify", storageframework.GetDriverTimeouts(driver))
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	init := func(ctx context.Context, createVolumeWithVAC bool) {
+		l = local{}
+
+		l.config = driver.PrepareTest(ctx, f)
+		vacDriver, _ := driver.(storageframework.VolumeAttributesClassTestDriver)
+		l.vac = vacDriver.GetVolumeAttributesClass(ctx, l.config)
+
+		if l.vac == nil {
+			e2eskipper.Skipf("Driver %q returned nil VolumeAttributesClass - skipping", driver.GetDriverInfo().Name)
+		}
+
+		ginkgo.By("Creating VolumeAttributesClass")
+		_, err := f.ClientSet.StorageV1alpha1().VolumeAttributesClasses().Create(ctx, l.vac, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "While creating VolumeAttributesClass")
+
+		ginkgo.By("Creating volume")
+		testVolumeSizeRange := v.GetTestSuiteInfo().SupportedSizeRange
+		if createVolumeWithVAC {
+			l.resource = storageframework.CreateVolumeResourceWithVAC(ctx, driver, l.config, pattern, testVolumeSizeRange, &l.vac.Name)
+		} else {
+			l.resource = storageframework.CreateVolumeResource(ctx, driver, l.config, pattern, testVolumeSizeRange)
+		}
+	}
+
+	cleanup := func(ctx context.Context) {
+		var errs []error
+		if l.resource != nil {
+			ginkgo.By("Deleting VolumeResource")
+			errs = append(errs, l.resource.CleanupResource(ctx))
+			l.resource = nil
+		}
+
+		if l.vac != nil {
+			ginkgo.By("Deleting VAC")
+			CleanupVAC(ctx, l.vac, f.ClientSet, vacCleanupWaitPeriod)
+			l.vac = nil
+		}
+
+		framework.ExpectNoError(errors.NewAggregate(errs), "While cleaning up")
+	}
+
+	ginkgo.It("should create a volume with VAC", func(ctx context.Context) {
+		init(ctx, true /* volume created with VAC */)
+		ginkgo.DeferCleanup(cleanup)
+
+		ginkgo.By("Creating a pod with dynamically provisioned volume")
+		podConfig := e2epod.Config{
+			NS:            f.Namespace.Name,
+			PVCs:          []*v1.PersistentVolumeClaim{l.resource.Pvc},
+			SeLinuxLabel:  e2epod.GetLinuxLabel(),
+			NodeSelection: l.config.ClientNodeSelection,
+			ImageID:       e2epod.GetDefaultTestImageID(),
+		}
+		pod, err := e2epod.CreateSecPodWithNodeSelection(ctx, f.ClientSet, &podConfig, f.Timeouts.PodStart)
+		ginkgo.DeferCleanup(e2epod.DeletePodWithWait, f.ClientSet, pod)
+		framework.ExpectNoError(err, "While creating test pod with VAC")
+
+		createdPVC, err := f.ClientSet.CoreV1().PersistentVolumeClaims(f.Namespace.Name).Get(ctx, l.resource.Pvc.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "While getting created PVC")
+		// HACK: Workaround CurrentVolumeAttributesClassName is not set by external-provisioner
```

Comment:
add a todo with link to github issue

Reply:
No longer necessary because we don't expect `external-provisioner` to set the status as discussed above, resolving